### PR TITLE
Feature/cds theme updates

### PR DESCRIPTION
**Pull request by peetritense**
Multiple changes to styling including FormIO and case-detail page.

### CODE REVIEW

File: projects/valtimo/components/assets/css/compatibility.scss [hunk 1]
```diff
@@ -249,3 +538,74 @@ body {
     }
   }
 }
+
+/* FormIO */
```
Comment (mbritense) [inline projects/valtimo/components/assets/css/compatibility.scss:558]: remove commented styling

File: projects/valtimo/components/src/lib/services/cds-theme.service.ts [hunk 2]
```diff
@@ -46,31 +53,53 @@ export class CdsThemeService implements OnDestroy {
...
-            break;
-          case SelectableCarbonTheme.G90:
-            this.setThemeOnDocument(SelectableCarbonTheme.G90);
-            break;
-        }
-      }
+    this._subscriptions.add(
+      this.preferredTheme$
+        .pipe(
+          switchMap((preferredTheme: SelectableCarbonTheme) =>
```
Comment (mbritense) [inline projects/valtimo/components/src/lib/services/cds-theme.service.ts]: Why this refactor? It seems a lot less readable to me

File: projects/valtimo/components/src/lib/services/cds-theme.service.ts [hunk 3]
```diff
@@ -46,31 +53,53 @@ export class CdsThemeService implements OnDestroy {
...
+                ? SelectableMonacoTheme.VS
+                : SelectableMonacoTheme.VSDARK;
+          }
+          this.setThemeOnDocument(documentTheme);
+          this._editorTheme$.next(editorTheme);
+        })
+    );
+  }
+
+  private openMonacoThemeSubscription(): void {
```
Comment (mbritense) [inline projects/valtimo/components/src/lib/services/cds-theme.service.ts]: This logic should be moved to the EditorService, where also the editor instance should be saved

File: projects/valtimo/components/src/lib/services/cds-theme.service.ts [hunk 4]
```diff
@@ -46,31 +53,53 @@ export class CdsThemeService implements OnDestroy {
...
+        })
     );
   }
 
   private setThemeOnDocument(theme: SelectableCarbonTheme): void {
     document.documentElement.setAttribute('data-carbon-theme', theme);
   }
 
+  private setMonacoEditorTheme(monacoTheme: SelectableMonacoTheme): void {
+    monaco?.editor?.setTheme(monacoTheme);
```
Comment (mbritense) [inline projects/valtimo/components/src/lib/services/cds-theme.service.ts]: this should work with an instance in EditorService and this logic to set theme should be moved there

File: projects/valtimo/components/src/lib/services/cds-theme.service.ts (hunk 2 above)
Comment (peetritense) [reply projects/valtimo/components/src/lib/services/cds-theme.service.ts]: Combination of adding functionality for Monaco editor, and removing dual initialisation of themes causing the Monaco editor theming to break.

File: projects/valtimo/analyse/src/lib/analyse-process-diagram/analyse-process-diagram.component.ts [hunk 5]
```diff
@@ -56,9 +57,13 @@ export class AnalyseProcessDiagramComponent implements OnInit, OnDestroy {
   public valueKey: string;
   private initialized = false;
 
-  constructor(private processService: ProcessService) {}
+  constructor(
+    private processService: ProcessService,
```
Comment (mbritense) [inline projects/valtimo/analyse/src/lib/analyse-process-diagram/analyse-process-diagram.component.ts]: private readonly

File: projects/valtimo/components/src/lib/components/timeline/timeline.component.html [hunk 6]
```diff
@@ -15,7 +15,9 @@
   -->
 
 <div>
-  <div id="no-results" class="p-5" *ngIf="items?.length === 0">No result found</div>
+  <div id="no-results" class="p-5" *ngIf="items?.length === 0">
+    <!--    <valtimo-no-results [description]="'Test'" [title]="'test2'"></valtimo-no-results>-->
```
Comment (mbritense) [inline projects/valtimo/components/src/lib/components/timeline/timeline.component.html]: remove commented code

File: projects/valtimo/components/src/lib/services/cds-theme.service.ts [hunk 7]
```diff
@@ -13,11 +13,12 @@
...
  * limitations under the License.
  */
-
 import {Injectable, OnDestroy} from '@angular/core';
 import {BehaviorSubject, combineLatest, Observable, Subscription} from 'rxjs';
-import {SelectableCarbonTheme} from '../models';
 import {filter} from 'rxjs/operators';
+import {CurrentCarbonTheme, SelectableCarbonTheme} from '../models';
+
+// declare const monaco: any;
```
Comment (mbritense) [inline projects/valtimo/components/src/lib/services/cds-theme.service.ts]: remove commented code

File: projects/valtimo/process-link/src/lib/components/form-link-process-diagram/form-link-process-diagram.component.ts [hunk 8]
```diff
@@ -53,10 +54,12 @@ export class FormLinkProcessDiagramComponent implements OnInit, OnDestroy {
 
   constructor(
     private processService: ProcessService,
-    private route: ActivatedRoute
+    private route: ActivatedRoute,
```
Comment (mbritense) [inline projects/valtimo/process-link/src/lib/components/form-link-process-diagram/form-link-process-diagram.component.ts]: private readonly

File: projects/valtimo/process-link/src/lib/components/form-link-process-diagram/form-link-process-diagram.component.ts [hunk 9]
```diff
@@ -53,10 +54,12 @@ export class FormLinkProcessDiagramComponent implements OnInit, OnDestroy {
 
   constructor(
     private processService: ProcessService,
```
Comment (mbritense) [inline projects/valtimo/process-link/src/lib/components/form-link-process-diagram/form-link-process-diagram.component.ts]: private readonly

**Review by mbritense — APPROVED** [review]

**Pull request merged**